File: app/Http/Controllers/TasksController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;

use App\Task;
use App\Project;
use App\Http\Requests;
use App\Http\Controllers\CrudController;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Facades\Auth;
use Illuminate\Support\Facades\Input;
use Carbon\Carbon;

class TasksController extends CrudController
{

    public function __construct()
    {
    }

    protected function getModel()
    {
        return Task::class;
    }

    protected function applyFilters(Request $request, $query) {
        if($request->has('projectId'))
           $query = $query->where('project_id', '=', $request->projectId);

        if($request->has('description'))
            $query = $query->where('description', 'like', '%'.$request->description.'%');

        if($request->has('done'))
           $query = $query->where('done', '=', $request->done);

        if($request->has('dateStart'))
           $query = $query->where('scheduled_to', '>=', Carbon::parse($request->dateEnd));

        if($request->has('dateEnd'))
           $query = $query->where('scheduled_to', '<=', Carbon::parse($request->dateEnd));
    }

    protected function beforeSearch(Request $request, $dataQuery, $countQuery) {
        $dataQuery->orderBy('description', 'asc');
    }

    protected function getValidationRules(Request $request, Model $obj)
    {
        $rules = [
            'description' => 'required|max:256',
            'priority' => 'required|min:1',
            'scheduled_to' => 'required'
        ];

        if ( strpos($request->route()->getName(), 'tasks.update') !== false ) {
            $rules['done'] = 'required';
        }

        return $rules;
    }

    public function beforeStore(Request $request, Model $obj)
    {
        $obj->done = false;
    }

    /**
     * Atualiza o status da tarefa
     */
    public function toggleDone(Request $request)
    {
        $task = Task::find($request->id);

        $this->validate($request, [
            'done' => 'required'
        ]);

        $task->done = $request->done;

        $task->save();

    }
}

File: Controllers/TasksController.cs
using StarterPack.Models;
using System.Linq;
using StarterPack.Core.Validation;
using FluentValidation;
using StarterPack.Core.Controllers;
using StarterPack.Core.Controllers.Attributes;
using Microsoft.EntityFrameworkCore;
using System;
using Microsoft.AspNetCore.Mvc;

namespace StarterPack.Controllers
{
    [Authorize]
    public class TasksController : CrudController<Task>
    {
        protected override void ApplyFilters(ref IQueryable<Task> query) {
            if( HasParameter("description") ) {
                query = query.Where(m => m.Description.ToLower().Contains(GetParameter("description").ToLower()));
            }
            if( HasParameter("id") ) {
                query = query.Where(m => m.Id == GetParameter<long>("id"));
            }
            if( HasParameter("projectId") ) {
                query = query.Where(m => m.Project.Id == GetParameter<long>("projectId"));
            }
            if( HasParameter("priority") ) {
                query = query.Where(m => m.Priority == GetParameter<int>("priority"));
            }
            if( HasParameter("done") ) {
                query = query.Where(m => m.Done == GetParameter<bool>("done"));
            }
            if( HasParameter("dateStart") ) {
                query = query.Where(m => m.ScheduledTo.Value.CompareTo( GetParameter<DateTime>("dateStart")) <= 0);
            }
            if( HasParameter("dateEnd") ) {
                query = query.Where(m => m.ScheduledTo.Value.CompareTo(GetParameter<DateTime>("dateEnd")) >= 0);
            }
        }

        protected override void BeforeSearch(ref IQueryable<Task> dataQuery, ref IQueryable<Task> countQuery)
        {
            dataQuery = dataQuery.OrderBy(m => m.Description);
        }

        protected override void SetValidationRules(Task model, ModelValidator<Task> validator)
        {
            validator.RuleFor(m => m.ProjectId).NotNull();
            validator.RuleFor(m => m.Description).NotEmpty().Length(10, 255);
            validator.RuleFor(m => m.Priority).NotNull().GreaterThanOrEqualTo(0).LessThanOrEqualTo(10);
            validator.RuleFor(m => m.ScheduledTo).NotNull().GreaterThanOrEqualTo(DateTime.UtcNow);
        }

        [HttpPut, Route("toggleDone")]
        public void ToggleDone([FromBody] Task attributes)
        {
            Task task = Task.Get(attributes.Id.Value);

            task.Done = attributes.Done;

            task.Save();
        }
    }
}
